cmd/gitcloner/integrationtests/clone_test.go

```diff
@@ -0,0 +1,465 @@
+package apply
+
+import (
+	"context"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/pem"
+	"io"
+	"net/http"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/go-git/go-git/v5"
+	"github.com/go-git/go-git/v5/config"
+	"github.com/go-git/go-git/v5/plumbing/object"
+	httpgit "github.com/go-git/go-git/v5/plumbing/transport/http"
+	"github.com/gogits/go-gogs-client"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	cp "github.com/otiai10/copy"
+	"github.com/rancher/gitjob/cmd/gitcloner/cmd"
+	"github.com/rancher/gitjob/cmd/gitcloner/gogit"
+	"github.com/testcontainers/testcontainers-go"
+	"golang.org/x/crypto/ssh"
+)
+
+const (
+	gogsUser = "test"
+	gogsPass = "pass"
+)
+
+var (
+	gogsClient   *gogs.Client
+	gogsCABundle []byte
+)
+
+var _ = Describe("Fleet apply gets content from git repo", func() {
+
+	var (
+		opts     *cmd.Options
+		private  bool
+		cloneErr error
+		tmp      string
+	)
+
+	When("Cloning an https repo", func() {
+		JustBeforeEach(func() {
+			container, err := createGogsContainer()
+			Expect(err).NotTo(HaveOccurred())
+			url, err := getHTTPSURL(context.Background(), container)
+			Expect(err).NotTo(HaveOccurred())
+			err = createRepo(url, "assets/repo", private)
+			Expect(err).NotTo(HaveOccurred())
+			opts.Repo = url + "/test/test-repo"
+			tmp = createTempFolder()
+			opts.Path = tmp
+		})
+
+		JustAfterEach(func() {
+			err := os.RemoveAll(tmp)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("Public repo that contains a README.md file", func() {
```

Comment:
```suggestion
		When("cloning a public repo that contains a README.md file", func() {
```
or if we don't want to repeat ourselves:
```suggestion
		When("the cloned repo is public and contains a README.md file", func() {
```